### PR TITLE
[E2E Test] Setup Test Network deployment

### PR DESCRIPTION
**Pull request by fleupold**
This mainly reuses the logic from the v1 deployment scripts (but adjusted to the v2 contracts and Uniswap). This will allow us to run the system against a local dev environment and simulate our desired demo use case as an end to end test.

### Test Plan
With ganache open (make sure it's set to at least Istanbul hardfork for the ChainId opcode), running 
```
(cd contracts; cargo run --bin deploy --features bin)
```

creates a correct deployment.

### CODE REVIEW

File: contracts/Cargo.toml [hunk 1]
```diff
@@ -13,21 +17,25 @@ default = []
...
 [dependencies]
-ethcontract = { version = "0.9",  default-features = false }
+ethcontract = { version = "0.9",  default-features = false, features = ["http"] }
 serde = "1.0"
 
 # [bin-dependencies]
 anyhow = { version = "1.0", optional = true }
 env_logger = { version = "0.8", optional = true }
-ethcontract-generate = { version = "0.9", optional = true }
+ethcontract-generate = { version = "0.9", optional = true}
```
Comment (nlordell) [inline contracts/Cargo.toml:34]: ubernit: 
```suggestion
ethcontract-generate = { version = "0.9", optional = true }
```

File: contracts/artifacts/UniswapV2Factory.json [hunk 2]
```diff
@@ -0,0 +1 @@
...81900360200190a35050565b6dffffffffffffffffffffffffffff166e0100000000000000000000000000000290565b60006dffffffffffffffffffffffffffff82167bffffffffffffffffffffffffffffffffffffffffffffffffffffffff841681612ab457fe5b049392505050565b80820182811015610df657604080517f08c379a000000000000000000000000000000000000000000000000000000000815260206004820152601460248201527f64732d6d6174682d6164642d6f766572666c6f77000000000000000000000000604482015290519081900360640190fdfe556e697377617056323a20494e53554646494349454e545f4f55545055545f414d4f554e54556e697377617056323a20494e53554646494349454e545f494e5055545f414d4f554e54556e697377617056323a20494e53554646494349454e545f4c4951554944495459556e697377617056323a20494e53554646494349454e545f4c49515549444954595f4255524e4544556e697377617056323a20494e53554646494349454e545f4c49515549444954595f4d494e544544a265627a7a723158207dca18479e58487606bf70c79e44d8dee62353c9ee6d01f9a9d70885b8765f2264736f6c63430005100032454950373132446f6d61696e28737472696e67206e616d652c737472696e672076657273696f6e2c75696e7432353620636861696e49642c6164647265737320766572696679696e67436f6e747261637429a265627a7a723158202760f92d7fa1db6f5aa16307bad65df4ebcc8550c4b1f03755ab8dfd830c178f64736f6c63430005100032"}
```
Comment (nlordell) [inline contracts/artifacts/UniswapV2Factory.json:1]: I noticed this includes the manually added `0x`. Do you think its worth waiting for `ethcontract 0.9.1` so manual modification of the JSON is not needed?

File: contracts/src/bin/deploy.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    let http = Http::new(NODE_URL)?;
+    let web3 = Web3::new(http);
+
+    log::info!("checking connection to local test node {}", NODE_URL);
+    wait_for_node(&web3).await?;
+
+    let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+
+    macro_rules! deploy {
+            ($contract:ident) => { deploy!($contract ()) };
```
Comment (nlordell) [inline contracts/src/bin/deploy.rs:39]: ubernit: Indentation is off by one. Sometimes `rustfmt` struggles with macros.

**Review by nlordell — APPROVED** [review]
Looks great!

**Review by e00E — APPROVED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 [workspace]
 members = [
     "contracts",
+    "e2e",
```
Comment (e00E) [inline Cargo.toml:4]: I think CI fails because this crate doesn't exist yet.

**Review by josojo — APPROVED** [review]
Progress!

**Pull request merged**